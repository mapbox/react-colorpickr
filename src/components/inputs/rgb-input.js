'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import NumberInput from './number-input';

class RGBInput extends React.PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    theme: PropTypes.object.isRequired,
    value: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired
  };

  render() {
    return <NumberInput {...this.props} min={0} max={255} />;
  }
}

export default RGBInput;
