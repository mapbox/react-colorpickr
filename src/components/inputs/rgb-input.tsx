import React from 'react';
import { NumberInput } from './number-input.tsx';

function RGBInput(props) {
  return <NumberInput {...props} min={0} max={255} />;
}

export { RGBInput };
