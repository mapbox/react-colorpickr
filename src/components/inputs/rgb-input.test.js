import React from 'react';
import { render } from '@testing-library/react';
import { RGBInput } from './rgb-input';

describe('RGBInput', () => {
  test('renders', () => {
    const props = {
      id: 'r',
      value: 200,
      theme: {},
      onChange: jest.fn()
    };

    const { baseElement } = render(<RGBInput {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
