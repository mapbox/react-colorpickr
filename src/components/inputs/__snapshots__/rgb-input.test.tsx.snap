// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RGBInput renders 1`] = `
<body>
  <div>
    <div>
      <label
        for="r"
      >
        r
      </label>
      <input
        id="r"
        max="255"
        min="0"
        step="1"
        type="number"
        value="200"
      />
    </div>
  </div>
</body>
`;
